rfcs/2023-12-25-proxy-implementation.md

```diff
@@ -0,0 +1,108 @@
+# SMP proxy implementation concerns
+
+* SMP proxy traffic should be indistinguishable from ordinary relay operation.
+* Proxy servers should be implemented by the same binary to prevent maintenance bloat.
+
+This leads to the proxy functionality added directly to the SMP protocol, raising its version.
+
+## Involved parties
+
+### Client
+
+> Client, from the servers' point of view, means the agent running inside a client app. An app/agent protocol is unchanged.
```

Comment:
user/agent protocol will change too, as the user should have control whether proxy is used for all, untrusted only or none of the relays.